R/locate_credentials.R

```diff
@@ -184,32 +219,7 @@ function(key = NULL,
                 }
             }
         }
-        # now find region, with fail safes
-        if (!is.null(region) && region != "") {
-            region <- region
-            if (isTRUE(verbose)) {
-                message(sprintf("Using user-supplied value for AWS Region ('%s')", region))
-            }
-        } else if (!is.null(env$region) && env$region != "") {
-            region <- env$region
-            if (isTRUE(verbose)) {
-                message(sprintf("Using Environment Variable 'AWS_DEFAULT_REGION' for AWS Region ('%s')", region))
-            }
-        } else {
-            # check instance metadata for region
-            reg <- try(aws.ec2metadata::instance_document()$region, silent = TRUE)
-            if (!inherits(reg, "try-error") && !is.null(reg) && reg != "") {
-                region <- reg
-                if (isTRUE(verbose)) {
-                    message(sprintf("Using EC2 Instance Metadata for AWS Region ('%s')", region))
-                }
-            } else {
-                region <- default_region
-                if (isTRUE(verbose)) {
-                    message(sprintf("Using default value for AWS Region ('%s')", region))
-                }
-            }
-        }
+        region <- get_region(region, default_region)
```

Comment:
Can you undo this change? I don't think we need this as a separate function (and the function as proposed won't work because it doesn't pass the `env` variable forward).